src/ideal/ideal.jl

```diff
@@ -124,13 +128,13 @@ coefficient ring were the rational numbers for example, the coefficients of the
 polynomials would be reduced to lowest terms.
 """
 function normalize!(I::sideal)
-   libSingular.id_Normalize(I.ptr, base_ring(I).ptr)
+   GC.@preserve I libSingular.id_Normalize(I.ptr, base_ring(I).ptr)
```

Comment:
Are we sure that `base_ring(I)` is preserved when `I` is preserved? (sorry if this is obvious, I don't have the structures in mind).

Reply:
I think it is, but I will rewrite those places for extra security.